go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/argoproj/pkg
 
-go 1.14
+go 1.21
```

Comment:
Is it necessary to bump the minimum supported Go version here? I mentioned this in https://github.com/argoproj/pkg/pull/707#discussion_r1704193810 that CI and local use is one thing, but this actually impacts consumers / is a breaking change

Reply:
I'm pretty sure `go mod tidy` did it for me.

Reply:
yea, even running make lint will update the go.mod file.

Reply:
CI fails without the go.mod update. I reverted the change

Reply:
That looks like because the [`toolchain` config](https://go.dev/doc/toolchain) you added below requires Go 1.21+

Reply:
Well I didn't add it. golang automatically added it.

Reply:
Oh. Huh that was my first time seeing it and we don't have it [in Workflows](https://github.com/argoproj/argo-workflows/blob/ca6c4144ced98e111d77111fc696984739607694/go.mod#L3)

I guess your version of Go will auto-add it? 🤔
Ironically you can't specify the 1.14 toolchain since that's older than 1.21

Reply:
yeah, my local golang is 1.23 from homebrew.